### PR TITLE
Use config value over default if not in env

### PR DESCRIPTION
**Pull request by marcoderama**
This is a proposed fix for https://github.com/ilyakaznacheev/cleanenv/issues/40.

This behaves the way I expected, given this line of the documentation:
> 3. if no value was found on the first two steps, the field will be filled with the default value (env-default tag) if it is set.

in https://github.com/ilyakaznacheev/cleanenv#read-configuration.

I'm no reflect wizard, so I suggest a careful review.  It works in my testing though.

### CODE REVIEW

File: cleanenv.go [hunk 1]
```diff
@@ -286,6 +286,10 @@ func readEnvVars(cfg interface{}, update bool) error {
 			}
 		}
 
+		if rawValue == nil && meta.fieldValue.IsZero() {
```
Comment (marcoderama) [inline cleanenv.go]: The IsZero() call requires Go 1.13 or better.

The point of this change is to
* use env value if supplied, or
* use value in config if supplied, or
* use default value if supplied.

**Comment by ilyakaznacheev** [comment]
I like it. Please add the logic described in the comments and push the changes. It also should trigger Travis check, because I've finally fixed it :)

Feel free to ping me if I don't answer for a long time, it seems that Github is not sending me some notifications.

**Comment by marcoderama** [comment]
> I like it. Please add the logic described in the comments and push the changes.

I'm not sure I follow.  Which comments?  If you mean here https://github.com/ilyakaznacheev/cleanenv/pull/41#discussion_r395398372, then it already does that.  Otherwise, I don't know which comments you're referring to.

> It also should trigger Travis check, because I've finally fixed it :)

Yes it does, and, not surprisingly, fails on the 1.12 and 1.11 builds since `IsZero()` only exists as of 1.13.  Assuming you need those to pass, this will have to be reworked.

**Comment by ilyakaznacheev** [comment]
@marcoderama I asked you to take the logic from the ` Value.IsZero()` method into your own method (e.g. `structMeta.isFieldValueZero() bool`) so that the code is compatible with versions 1.12 and 1.11.

I'm guessing it's gonna look something like this:

```go
func (sm *structMeta) isFieldValueZero() bool {
	switch sm. fieldValue.Kind() {
	case Bool:
		return !v.Bool()
	case Int, Int8, Int16, Int32, Int64:
		return v.Int() == 0
	case Uint, Uint8, Uint16, Uint32, Uint64, Uintptr:
		return v.Uint() == 0
	case Float32, Float64:
		return math.Float64bits(v.Float()) == 0
	case Complex64, Complex128:
		c := v.Complex()
		return math.Float64bits(real(c)) == 0 && math.Float64bits(imag(c)) == 0
	case Array:
		for i := 0; i < v.Len(); i++ {
			if !v.Index(i).IsZero() {
				return false
			}
		}
		return true
	case Chan, Func, Interface, Map, Ptr, Slice, UnsafePointer:
		return v.IsNil()
	case String:
		return v.Len() == 0
	case Struct:
		for i := 0; i < v.NumField(); i++ {
			if !v.Field(i).IsZero() {
				return false
			}
		}
		return true
	default:
		// This should never happens, but will act as a safeguard for
		// later, as a default value doesn't makes sense here.
		panic(&ValueError{"reflect.Value.IsZero", v.Kind()})
	}
}
```

So that can be used as 

```go
if rawValue == nil && meta.isFieldValueZero() {
...
}
```

and it will be compatible with 1.12 and 1.11.

That's my vision of the possible solution, but you're welcome offer yours.

**Pull request merged**